src/util/Keycloak.ts

```diff
@@ -7,4 +9,25 @@ const keycloak = Keycloak({
     clientId: process.env.REACT_APP_KEYCLOAK_CLIENTID!,
 });
 
+/**
+ * Refresh token minimum validity in seconds.
+ *
+ * Should be less than Access Token Lifespan in Keycloak.
+ */
+const KEYCLOAK_REFRESH_VALIDITY_TIME = 100;
+
+keycloak.onTokenExpired = () => {
+    keycloak.updateToken(KEYCLOAK_REFRESH_VALIDITY_TIME).then((refreshed) => {
+        if (!refreshed) {
+            keycloak.login({redirectUri: RoutingCls.buildFullUrl(Routes.dashboard)});
+        }
+    }).catch(() => {
+        keycloak.login({redirectUri: RoutingCls.buildFullUrl(Routes.dashboard)});
+    });
+};
+
+keycloak.onAuthRefreshError = () => {
+    keycloak.login({redirectUri: RoutingCls.buildFullUrl(Routes.dashboard)});
```

Comment:
This line repeats three times within ten LOCs, maybe worth extracting to a separate function.

Reply:
Good point, fixed.